Classes/WPEditorToolbarButton.m

```diff
@@ -59,93 +65,86 @@ - (void)setupAnimations
 	[self addTarget:self action:@selector(touchDown:) forControlEvents:UIControlEventTouchDown];
 	[self addTarget:self action:@selector(touchDragInside:) forControlEvents:UIControlEventTouchDragInside];
 	[self addTarget:self action:@selector(touchDragOutside:) forControlEvents:UIControlEventTouchDragOutside];
+    
+    self.circleScaleAnimation = [CABasicAnimation animationWithKeyPath:@"transform.scale"];
+    self.circleScaleAnimation.fromValue = [NSNumber numberWithFloat:0.5f];
+    self.circleScaleAnimation.toValue = [NSNumber numberWithFloat:1.4f];
+    [self.circleScaleAnimation setTimingFunction:[CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseIn]];
+    
+    self.circleOpacityAnimation = [CABasicAnimation animationWithKeyPath:@"opacity"];
+    self.circleOpacityAnimation.fillMode = kCAFillModeForwards;
+    self.circleOpacityAnimation.toValue = [NSNumber numberWithFloat:0.0];
+    [self.circleOpacityAnimation setTimingFunction:[CAMediaTimingFunction functionWithName:kCAMediaTimingFunctionEaseIn]];
+}
+
+- (void)startAnimation
+{
+    CGFloat circleLineWidth = 1.0f;
+    CGRect circleRect = CGRectInset(self.bounds, circleLineWidth / 2, circleLineWidth / 2);
+    CGPoint drawPoint = CGPointMake(CGRectGetMidX(circleRect), CGRectGetMidY(circleRect));
+    CAShapeLayer *circleLayer = [CAShapeLayer layer];
+    int radius = IS_IPAD ? TouchAnimationCircleRadiusiPad : TouchAnimationCircleRadius;
+    circleLayer.path = [UIBezierPath bezierPathWithArcCenter:CGPointZero radius:radius startAngle:0 endAngle:M_PI*2 clockwise:NO].CGPath;
+    circleLayer.position = drawPoint;
+    circleLayer.fillColor =  [[WPStyleGuide greyLighten10] CGColor];
+    circleLayer.strokeColor =  [[WPStyleGuide greyLighten10] CGColor];
+    circleLayer.lineWidth = 1.0f;
+    circleLayer.opacity = TouchAnimationInitialOpacity;
+    [self.layer addSublayer:circleLayer];
+    
+    CAAnimationGroup * group =[CAAnimationGroup animation];
+    group.removedOnCompletion=NO; group.fillMode=kCAFillModeForwards;
+    group.animations =[NSArray arrayWithObjects:self.circleScaleAnimation, self.circleOpacityAnimation, nil];
+    group.duration = TouchAnimationDuration;
+    group.repeatCount = 0.0f;
+    group.removedOnCompletion = NO;
```

Comment:
`removedOnCompletion` is set twice?  It looks like all animations are removed in `animationDidStop:finished:` anyway so does this need to be set to NO at all?